plugins/vli/fu-vli-pd-device.c

```diff
@@ -697,6 +703,12 @@ fu_vli_pd_device_detach(FuDevice *device, FuProgress *progress, GError **error)
 					error))
 		return FALSE;
 
+	/* check if device has skip-rom-mode flag */
+	if (fu_vli_device_get_kind(FU_VLI_DEVICE(device)) != FU_VLI_DEVICE_KIND_VL103 &&
+	    fu_device_has_private_flag(device, FU_VLI_PD_DEVICE_FLAG_SKIPS_ROM)) {
```

Comment:
@memily why the ` != FU_VLI_DEVICE_KIND_VL103` here out of interest?

Reply:
VL103 hardware does not support the feature to update in firmware mode. Amended comment to reflect answer.